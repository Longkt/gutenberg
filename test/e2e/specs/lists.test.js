/**
 * Internal dependencies
 */
import {
	clickBlockAppender,
	getEditedPostContent,
	newPost,
	pressTimes,
	convertBlock,
	pressWithModifier,
	insertBlock,
} from '../support/utils';

describe( 'Lists', () => {
	beforeEach( async () => {
		await newPost();
	} );

	it( 'can be created by using an asterisk at the start of a paragraph block', async () => {
		// Create a block with some text that will trigger a list creation.
		await clickBlockAppender();
		await page.keyboard.type( '* A list item' );

		// Create a second list item.
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'Another list item' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by typing an asterisk in front of text of a paragraph block', async () => {
		// Create a list with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await pressTimes( 'ArrowLeft', 4 );
		await page.keyboard.type( '* ' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by using a number at the start of a paragraph block', async () => {
		// Create a block with some text that will trigger a list creation.
		await clickBlockAppender();
		await page.keyboard.type( '1) A list item' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by typing "/list"', async () => {
		// Create a list with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/list' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'Iâ€™m a list' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by converting a paragraph', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await convertBlock( 'List' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by converting multiple paragraphs', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'one' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'two' );
		await page.keyboard.down( 'Shift' );
		await page.click( '[data-type="core/paragraph"]' );
		await page.keyboard.up( 'Shift' );
		await convertBlock( 'List' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by converting a paragraph with line breaks', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'one' );
		await pressWithModifier( 'Shift', 'Enter' );
		await page.keyboard.type( 'two' );
		await convertBlock( 'List' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be converted to a paragraphs', async () => {
		await insertBlock( 'List' );
		await page.keyboard.type( 'one' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'two' );
		await convertBlock( 'Paragraph' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
