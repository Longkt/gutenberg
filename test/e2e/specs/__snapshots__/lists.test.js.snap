// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists can be converted to a quote 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>one</p><p>two</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Lists can be converted to paragraphs 1`] = `
"<!-- wp:paragraph -->
<p>one</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>two</p>
<!-- /wp:paragraph -->"
`;

exports[`Lists can be created by converting a paragraph 1`] = `
"<!-- wp:list -->
<ul><li>test</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by converting a paragraph with line breaks 1`] = `
"<!-- wp:list -->
<ul><li>one</li><li>two</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by converting a quote 1`] = `
"<!-- wp:list -->
<ul><li>one</li><li>two</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by converting multiple paragraphs 1`] = `
"<!-- wp:list -->
<ul><li>one</li><li>two</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by typing "/list" 1`] = `
"<!-- wp:list -->
<ul><li>Iâ€™m a list</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by typing an asterisk in front of text of a paragraph block 1`] = `
"<!-- wp:list -->
<ul><li>test</li></ul>
<!-- /wp:list -->"
`;

exports[`Lists can be created by using a number at the start of a paragraph block 1`] = `
"<!-- wp:list {\\"ordered\\":true} -->
<ol><li>A list item</li></ol>
<!-- /wp:list -->"
`;

exports[`Lists can be created by using an asterisk at the start of a paragraph block 1`] = `
"<!-- wp:list -->
<ul><li>A list item</li><li>Another list item</li></ul>
<!-- /wp:list -->"
`;
